functions/src/logic/ingestUnityVersions/scrapeVersions.ts

```diff
@@ -1,53 +1,33 @@
-import { getDocumentFromUrl } from '../utils/get-document-from-url';
 import { EditorVersionInfo } from '../../model/editorVersionInfo';
+import { searchChangesets, SearchMode } from 'unity-changeset';
 
-const UNITY_ARCHIVE_URL = 'https://unity.com/releases/editor/archive';
-const unity_version_regex = /unityhub:\/\/(\d+)\.(\d+)\.(\d+[a-zA-Z]\d+)\/(\w+)/g;
+const unity_version_regex = /^(\d+)\.(\d+)\.(\d+)([a-zA-Z]+)(-?\d+)$/;
```

Comment:
I see that the version regex is now different. I assume what we retrieve from GraphQL api is slightly different from what we used with the webpage before so it's most likely ok.

Here's a table with **Unity versions** as rows and their validity against each regex as columns:

| **Unity Version**         | **Previous regex** (`unityhub:\/\/(\d+)\.(\d+)\.(\d+[a-zA-Z]\d+)\/(\w+)`) | **New regex** (`/^(\d+)\.(\d+)\.(\d+)([a-zA-Z]+)(-?\d+)$/`) |
|----------------------------|-------------------------------------------------------------------|----------------------------------------------------------|
| `unityhub://2023.1.15a1/release` | **Valid**                                                              | **Invalid** (URL format not supported)                  |
| `2023.1.15a-1`             | **Invalid** (missing `unityhub://` prefix and identifier)                    | **Valid**                                                |
| `unityhub://2022.3.12b2/alpha`  | **Valid**                                                              | **Invalid**                                              |
| `2021.2.0f1`               | **Invalid** (missing URL structure)                                         | **Valid**                                                |
| `unityhub://2019.4.28c2/beta`   | **Valid**                                                              | **Invalid**                                              |
| `2020.1.0b12`              | **Invalid** (missing URL structure)                                         | **Valid**                                                |
| `unityhub://2023.1.15/release`  | **Invalid** (missing alpha/beta tag in patch version)                   | **Invalid**                                              |
| `2023.1.15`                | **Invalid** (no URL structure)                                              | **Invalid** (missing alpha/beta/final tag and revision)  |
| `2023.1.15a1`              | **Invalid** (no URL structure or identifier)                                | **Invalid** (missing optional suffix like `-1`)          |
| `unityhub://2020.3.10b2`   | **Invalid** (missing identifier like `release` or `beta`)                   | **Invalid**                                              |

### Key Observations:
1. **Regex 1** is **strict** and only works for `unityhub://` URL structures with all required components (version + identifier).
2. **Regex 2** is **flexible** for standalone version strings and doesn’t use URL-based Unity Hub links.

That looks good to me :+1: